src/com/pentalog/pentastagiu/homework2/Basket.java

```diff
@@ -0,0 +1,19 @@
+package com.pentalog.pentastagiu.homework2;
+
+public class Basket {
+    private double price;
+    private Customer customer;
+
+    public Basket(Customer customer) {
+        this.customer = customer;
+        this.price = 0;
+    }
+
+    public void addProduct(Product product) {
+        this.price += product.getPrice();
+    }
+
+    public double getPrice() {
+        return this.price  * this.customer.getDiscount();
```

Comment:
ce se intampla daca cineva creaza un basket asa: new Basket(null)? ce se va intampla cand vei invoca metoda getPrice()? o exceptie e aruncata, ceea ce sugereaza ca trebuie adaugate validari suplimentare(ori in metoda asta, ori in constructor: cu cat mai repede, cu atat mai bine)